EDMarketConnector.py

```diff
@@ -730,27 +775,41 @@ def crewroletext(role):
                 self.updater.setAutomaticUpdatesCheck(True)
                 logger.info('Monitor: Enable WinSparkle automatic update checks')
 
-    # cAPI auth
-    def auth(self, event=None):
+    def auth(self, event=None) -> None:
+        """
+        Handle Frontier auth callback.
+
+        This is the callback function for the CompanionAuthEvent Tk event.
+        It is triggered by the event() function of class GenericProtocolHandler
+        in protocol.py.
+        """
         try:
             companion.session.auth_callback()
             # Successfully authenticated with the Frontier website
             self.status['text'] = _('Authentication successful')
             if platform == 'darwin':
                 self.view_menu.entryconfigure(0, state=tk.NORMAL)  # Status
                 self.file_menu.entryconfigure(0, state=tk.NORMAL)  # Save Raw Data
+
             else:
                 self.file_menu.entryconfigure(0, state=tk.NORMAL)  # Status
                 self.file_menu.entryconfigure(1, state=tk.NORMAL)  # Save Raw Data
+
         except companion.ServerError as e:
             self.status['text'] = str(e)
+
         except Exception as e:
             logger.debug('Frontier CAPI Auth:', exc_info=e)
             self.status['text'] = str(e)
+
         self.cooldown()
 
-    # Handle Status event
-    def dashboard_event(self, event):
+    def dashboard_event(self, event) -> None:
```

Comment:
Can we assume that event is at least a `Mapping[str, Any]` here?

Reply:
That results in:

```py
EDMarketConnector.py:340: error: Argument 2 to "bind_all" of "Misc" has incompatible type "Callable[[Mapping[str, Any]], None]"; expected "Optional[Callable[[Event[Misc]], Optional[Literal['break']]]]"
```

Reply:
Oh, thats wonderful. Nevermind then